### PR TITLE
Deduplication  : done

### PR DESCRIPTION
**Pull request by truthfool**

### CODE REVIEW

File: openg2p_disbursement/models/openg2p_disbursement_slip.py [hunk 1]
```diff
@@ -618,3 +621,68 @@ def get_disbursement_line_total(self, code):
             return line[0].total
         else:
             return 0.0
+
+    def create_bulk_transfer(self):
+        query = """SELECT p.total, p.currency_id, c.sanitized_acc_number
+                        FROM public.openg2p_disbursement_advice AS p  
+                        LEFT JOIN public.openg2p_disbursement_advice FROM public.res_partner_bank AS c  
+                        ON p.id=c.id  
+                        INTO OUTFILE 'accounts.csv' """
```
Comment (SubhamPramanik) [inline openg2p_disbursement/models/openg2p_disbursement_slip.py]: What are the CSV file headers?

File: openg2p_disbursement/models/openg2p_disbursement_slip.py [hunk 2]
```diff
@@ -618,3 +621,68 @@ def get_disbursement_line_total(self, code):
...
+
+        headers = {
+            "Content-Type": "multipart/form-data",
+        }
+
+        files = {
+            "data": ("accounts.csv", open("accounts.csv", "rb")),
+            "note": (None, "Bulk transfers"),
+            "checksum": (None, str(self.hash_generate())),
```
Comment (SubhamPramanik) [inline openg2p_disbursement/models/openg2p_disbursement_slip.py]: Is hash_generator supposed to have an argument which takes the file/filename?

File: openg2p_registration/models/openg2p_registration.py [hunk 3]
```diff
@@ -581,6 +590,123 @@ def create_beneficiary_from_registration(self):
...
+                "postal_code": str(self.zip),
+                "identity": str(self.identity_passport),
+                "bank": str(self.bank_account_id.bank_id.name),
+                "bank_account": str(self.bank_account_id.sanitized_acc_number),
+                "emergency_contact_name": str(self.emergency_contact),
+                "emergency_contact_phone": str(self.emergency_phone),
+            }
+        )
+
+        delete_url = "http://localhost:8080/index/" + str(idr)
```
Comment (SubhamPramanik) [inline openg2p_registration/models/openg2p_registration.py]: Move and break the URL into smalled chunk and put it in a class/global level variable. 
Deduplication_host = http://localhost:8080
Index_path = /index
search_path...
etc

File: openg2p_registration/models/openg2p_registration.py [hunk 4]
```diff
@@ -581,6 +590,123 @@ def create_beneficiary_from_registration(self):
...
+            "identity": str(self.identity_passport),
+            "bank": str(self.bank_account_id.bank_id.name),
+            "bank_account": str(self.bank_account_id.sanitized_acc_number),
+            "emergency_contact_name": str(self.emergency_contact),
+            "emergency_contact_phone": str(self.emergency_phone),
+        }
+        # Deleting null fields
+        new_data = self.del_none(data)
+        print(new_data)
+        url_endpoint = "http://localhost:8080/index"
```
Comment (SubhamPramanik) [inline openg2p_registration/models/openg2p_registration.py]: Same here

File: openg2p_registration/models/openg2p_registration.py [hunk 5]
```diff
@@ -581,6 +590,123 @@ def create_beneficiary_from_registration(self):
...
+                "postal_code": str(self.zip),
+                "dob": str(self.birthday),
+                "identity": str(self.identity_passport),
+                "bank": str(self.bank_account_id.bank_id.name),
+                "bank_account": str(self.bank_account_id.sanitized_acc_number),
+                "emergency_contact_name": str(self.emergency_contact),
+                "emergency_contact_phone": str(self.emergency_phone),
+            }
+        }
+        new_data = self.del_none(search_data)
```
Comment (SubhamPramanik) [inline openg2p_registration/models/openg2p_registration.py]: Rename these variables - new_data. my_data etc. Make it a more suitable naming.

File: openg2p_registration/views/openg2p_registration.xml [hunk 6]
```diff
@@ -22,7 +22,7 @@
             </field>
         </record>
 
-        <record id="openg2p_registration_view_tree_activity" model="ir.ui.view">
+        <!-- <record id="openg2p_registration_view_tree_activity" model="ir.ui.view">
```
Comment (SubhamPramanik) [inline openg2p_registration/views/openg2p_registration.xml]: Why did we comment this code?

File: openg2p_disbursement/models/openg2p_disbursement_slip.py (hunk 1 above)
Comment (truthfool) [reply openg2p_disbursement/models/openg2p_disbursement_slip.py]: Thsi is the earlier written code i need to comment it out.All the logic for single and batch transaction is in openg2p_disbursement_batch_transaction,openg2p_disbursement_single_transaction,openg2p_disbursement_main.

File: openg2p_disbursement/models/openg2p_disbursement_slip.py (hunk 2 above)
Comment (truthfool) [reply openg2p_disbursement/models/openg2p_disbursement_slip.py]: Same goes for here i need to comment this out or remove it from Salton's code.

File: openg2p_registration/models/openg2p_registration.py (hunk 3 above)
Comment (truthfool) [reply openg2p_registration/models/openg2p_registration.py]: OK i will make necessary changes.

File: openg2p_registration/models/openg2p_registration.py (hunk 4 above)
Comment (truthfool) [reply openg2p_registration/models/openg2p_registration.py]: OK.

File: openg2p_registration/models/openg2p_registration.py (hunk 5 above)
Comment (truthfool) [reply openg2p_registration/models/openg2p_registration.py]: Sure, will rename it.

File: openg2p_registration/views/openg2p_registration.xml (hunk 6 above)
Comment (truthfool) [reply openg2p_registration/views/openg2p_registration.xml]: It showed error while upgrading registration module.

**Pull request closed without merging**